Source/JavaScriptCore/jit/JITBitOrGenerator.cpp

```diff
@@ -42,28 +42,33 @@ void JITBitOrGenerator::generateFastPath(CCallHelpers& jit)
         
         // Try to do intVar | intConstant.
         m_slowPathJumpList.append(jit.branchIfNotInt32(var));
-        
-        jit.moveValueRegs(var, m_result);
+
         if (constOpr.asConstInt32()) {
 #if USE(JSVALUE64)
+#if CPU(ARM64)
+            jit.or64(CCallHelpers::TrustedImm64(static_cast<uint64_t>(static_cast<uint32_t>(constOpr.asConstInt32()))), var.payloadGPR(), m_result.payloadGPR());
+#else
+            jit.moveValueRegs(var, m_result);
             jit.or32(CCallHelpers::Imm32(constOpr.asConstInt32()), m_result.payloadGPR());
             jit.or64(GPRInfo::numberTagRegister, m_result.payloadGPR());
+#endif
 #else
```

Comment:
Let's introduce `or64(Imm32 ...` in ternary form (like we have for `and64`) and use that instead?  Seems it would make this code cleaner and more consistent with the `and` and `xor` cases, plus other 64-bit platforms with capabilities similar to ARM64 (e.g. RISCV64) will also pick up the same benefit.

Reply:
It requires Imm64 blinding on non ARM64, and it is substantially complicated.
Given that this is only beneficial to x64 and this place, I would like to make it just a TODO for the future.